### PR TITLE
Add docker support for development and testing

### PR DESCRIPTION
**Pull request by devertap**
In this pull request, I've added a base elixir image with some minor dependencies.  Then it is wired up using docker-compose in conjunction with a standard postgres:9.5 image.  

I then documented it via the README.  You'll notice I changed the config for the local PG_USER/PASSWORD.  This was to make them the same and accessible via the dev.exs and test.exs.

Hope it helps.

### CODE REVIEW

**Comment by mbramson** [comment]
Hey, thanks for the PR! I'll take a look at this this week. I have a few compatibility concerns since this app is currently deployed to heroku. Also take a look at those CI failures. Tests will have to be passing before this is merged.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    - PORT=4000
+  volumes:
+    - .:/app
+  working_dir: /app
+  links:
+    - db
+db:
+  image: postgres:9.5
+  environment:
+    - POSTGRES_USER=postgres
```
Comment (Serneum) [inline docker-compose.yml]: I'm not familiar with docker-compose, but should this be `DB_ENV_POSTGRES_USER` like you had above, or is it automatically converted to that by the layout?

**Review by Serneum — COMMENTED** [review]
You may need to update travis.yml as well since it contains a settings POSTGRES_* variables. I think that's why the tests are failing

File: docker-compose.yml (hunk 1 above)
Comment (devertap) [reply docker-compose.yml]: Yeah this is the problem.  It cannot find the environment variables. If we are setting up the variables in test setup then it needs to be DB_ENV_POSTGRES_USER, DB_ENV_POSTGRES_PASSWORD.

**Comment by devertap** [comment]
Well, it looks like just changing that config wasn't enough.  When I get a little time I will take a look at this a little deeper.

**Comment by Serneum** [comment]
A `git pull origin master --rebase` is preferred over a merge commit

File: README.md [hunk 2]
```diff
@@ -19,11 +19,32 @@ Pairmotron is an app powered by [Phoenix](http://www.phoenixframework.org/) that
...
+  * docker-compose exec web mix deps.get
+  * docker-compose exec web mix ecto.create 
+  * docker-compose exec web mix ecto.migrate
+  * docker-compose exec web npm install
+  * docker-compose restart web
+
+### Run the Tests in Docker
+  * docker-compose run -e "MIX_ENV=test" web mix test 
+  * Only test a specific test file
+    * docker-compose run -e "MIX_ENV=test" web mix test/controllers/group_controller_test.exs
```
Comment (ericworkman) [inline README.md:44]: Typo in `web mix test/controllers/group_controller_test.exs`. Should be `web mix 
test test/controllers/group_controller_test.exs`

File: config/dev.exs [hunk 3]
```diff
@@ -39,5 +39,6 @@ config :pairmotron, Pairmotron.Repo,
   username: System.get_env("PG_USER"),
   password: System.get_env("PG_PASSWORD"),
   database: "pairmotron_dev",
-  hostname: "localhost",
+  hostname: System.get_env("PG_HOST"),
```
Comment (ericworkman) [inline config/dev.exs:42]: I suggest `System.get_env("PG_HOST") || "localhost"` for this file and test.exs. That allows for overriding but prevents existing developers from needing another env var.

**Review by ericworkman — CHANGES_REQUESTED** [review]
Thanks @devertap ! PR looks pretty good, just a couple of minor changes I'd suggest for now!

**Comment by devertap** [comment]
Closing and reopening in PR #171

**Pull request closed without merging**